### PR TITLE
feat(#5103): Adding nodeSelector to the operator helm chart

### LINKED ISSUE #5103: Include the operator node-selector in the helm chart
### Requirement

The kamel cli has the `--node-selector` option, but there's not such an option in the helm template (https://github.com/apache/camel-k/blob/main/helm/camel-k/templates/operator.yaml).
My idea is to add this option to the yaml file.

### Problem

It blocks me from using helm to install Camel K as I'm using then node selector option.

### Proposal

Add the node Selector as optional parameter to the chart template.

### Open questions

_No response_

### PR DESCRIPTION
**Pull request by lsergio**
This resolves #5103 

**Release Note**
```release-note
It is now possible to set the operator nodeSelector in the helm values file.
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: helm/camel-k/values.yaml [hunk 1]
```diff
@@ -28,6 +28,7 @@ operator:
   resources: {}
   securityContext: {}
   tolerations: []
+  nodeSelector: {}
```
Comment (gansheer) [inline helm/camel-k/values.yaml:31]: This should be added to the documentation here: https://github.com/apache/camel-k/blob/a98dec1f8f260c2e478543c1ed3f19113c0cc5d2/helm/camel-k/README.md?plain=1#L107

**Review by gansheer — COMMENTED** [review]
Thanks for the contribution!

File: helm/camel-k/values.yaml (hunk 1 above)
Comment (lsergio) [reply helm/camel-k/values.yaml:31]: Done!

**Review by gansheer — APPROVED** [review]

File: helm/camel-k/README.md [hunk 2]
```diff
@@ -15,7 +15,6 @@ To install the chart, first add the Camel K repository:
 
 ```bash
 $ helm repo add camel-k https://apache.github.io/camel-k/charts
-```
```
Comment (squakez) [inline helm/camel-k/README.md:18]: Maybe this is required.

Comment (lsergio) [reply helm/camel-k/README.md]: ooops, my bad. fixed it.

**Comment by squakez** [comment]
Merging this as the checks errors are unrelated.

**Pull request merged**